### DISCUSSION TITLE
NSSecureCoding support

### DISCUSSION THREAD

**Discussion by Juan2662** [invertase/react-native-firebase#7434] [Ideas] [answered]
Hi guys, I notice that some chunks in the code in iOS (like FirebaseMessaging) is using NSCoding protocol
Can we have support for NSSecureCoding?

**Comment by mikehardy** [comment]
I do not see NSCoding references anywhere in our library. The underlying firebase-ios-sdk has been updated as of version 7.11.0 to use NSSecureCoding. We are currently integrating with firebase-ios-sdk 10.17.0 so I do not believe this is a current issue if you are using current versions.

If I'm wrong, please post links to the lines of code in the repo (you can get them via github repo search on the web then grabbing the permalink from the line number).

**Reply by Juan2662** [reply]
I'm using 

`
"@react-native-firebase/analytics": "^18.3.2"
"@react-native-firebase/app": "^18.3.2"
"@react-native-firebase/crashlytics": "^18.3.2"
"@react-native-firebase/dynamic-links": "^18.3.2"
"@react-native-firebase/messaging": "^18.3.2"
"@react-native-firebase/perf": "^18.3.2"
"@react-native-firebase/remote-config": "^18.3.2"
`

When I run "pod install" I get this file that is using "NSCoding"
![image](https://github.com/invertase/react-native-firebase/assets/58581443/774d8c30-e358-4048-b52e-013f8208a87e)

**Reply by mikehardy** [reply, marked as answer]
Hi there!

1- That version is out of date, we are on 18.6.1 now, you are asking for support on old versions that may have the issue fixed already.
2- That file is in the firebase-ios-sdk and is thus not our responsibility here, we cannot change it. You may want to discuss in firebase-ios-sdk repository
3- I asked for a permalink - github is a *text* medium, images are difficult to handle as they cannot be easily referenced. As you head over to firebase-ios-sdk to inquire why they haven't changed this particular line, you will want to use the permalink to the line as I indicated https://github.com/firebase/firebase-ios-sdk/blob/aec227be2b101577e0b531a08ec057ef01e13703/FirebaseMessaging/Sources/Token/FIRMessagingTokenInfo.h#L28

**Reply by Juan2662** [reply]
Got it, Thank you so much @mikehardy
